### PR TITLE
Add placements to the list api of the table

### PR DESCRIPTION
**Pull request by zzbdatatom**
PR checklist:
- Did you run ClangFormat ? yes
- Did you run CheckStyle ? yes
- Did you check the comment / log / exception conventions in Engineering code process wiki page ? yes
- Did you import unnecessary headers ? no
- Did you surround `Daisy : starts/ends` for new code in existing ClickHouse code base ? yes

Please write user-readable short description of the changes:
1. Add placements to the list api of the table,  like this:
    "placements":[
        {
            "shard":0,
            "replicas":["hostname1", "hostname2" ]
        },
        {
            "shard":1,
            "replicas":["hostname3", "hostname4"]
        }
    ]

### CODE REVIEW

File: src/Server/RestRouterHandlers/RawstoreTableRestRouterHandler.cpp [hunk 1]
```diff
@@ -52,6 +52,32 @@ void RawstoreTableRestRouterHandler::buildTablesJSON(Poco::JSON::Object & resp,
         table_mapping_json.set("order_by_expression", table->sorting_key);
         table_mapping_json.set("partition_by_expression", table->partition_key);
 
+        const auto & catalog_service = CatalogService::instance(query_context);
```
Comment (hzhzhang) [inline src/Server/RestRouterHandlers/RawstoreTableRestRouterHandler.cpp]: Could we reuse the code block as they are identical.

Comment (zzbdatatom) [reply src/Server/RestRouterHandlers/RawstoreTableRestRouterHandler.cpp]: Yes

File: src/Server/RestRouterHandlers/TableRestRouterHandler.cpp [hunk 2]
```diff
@@ -247,6 +247,35 @@ void TableRestRouterHandler::buildColumnsJSON(Poco::JSON::Object & resp_table, c
...
+        auto range = nodes.equal_range(it->first);
+        Poco::JSON::Array replicas;
+        while (range.first != range.second)
+        {
+            replicas.add(range.first++->second);
+        }
+        placement.set("replicas", replicas);
+        placements.add(placement);
+    }
+    resp_table.set("placements", placements);
```
Comment (chenziliang) [inline src/Server/RestRouterHandlers/TableRestRouterHandler.cpp]: `placements` is internal. Let's use `shards`

Comment (zzbdatatom) [reply src/Server/RestRouterHandlers/TableRestRouterHandler.cpp]: fixed

**Pull request merged**